File: ios/screenusageapp/ScreenTimeModule.m
#import "ScreenTimeModule.h"
#import <React/RCTLog.h>

@implementation ScreenTimeModule

RCT_EXPORT_MODULE();

RCT_EXPORT_METHOD(getUsage:(RCTPromiseResolveBlock)resolve reject:(RCTPromiseRejectBlock)reject)
{
    RCTLogInfo(@"getUsage called");

    // Here, you need to use Objective-C code equivalent to your Swift code
    // Unfortunately, DeviceActivityReport is not available in Objective-C,
    // so you'll need to bridge your Swift code.

    NSError *error = [NSError errorWithDomain:@"E_LOAD_FAILED" code:200 userInfo:nil];
    reject(@"E_LOAD_FAILED", @"Failed to load Device Activity Report", error);
}

@end

File: ios/screenusageapp/ScreenTimeModule.swift
import Foundation
import DeviceActivity
import React

@objc(ScreenTimeModule)
class ScreenTimeModule: NSObject {

    @objc
    func getUsage(_ resolve: @escaping RCTPromiseResolveBlock, reject: @escaping RCTPromiseRejectBlock) {
        print("getUsage called")
        let store = DeviceActivityReport()
        store.load { error in
            guard error == nil else {
                reject("E_LOAD_FAILED", "Failed to load Device Activity Report", error)
                return
            }

            var usageData: [String: TimeInterval] = [:]

            for application in store.allApplications {
                if let applicationData = store.applicationData(for: application) {
                    let totalUsage = applicationData.totalUsage
                    usageData[application] = totalUsage
                }
            }

            resolve(usageData)
        }
    }

    @objc
    static func requiresMainQueueSetup() -> Bool {
        return true
    }
}